### ISSUE TITLE
Robolectric tests fail - Unable to extract the trust manager v3.1.1

### ISSUE THREAD

**Issue by shubhamchaudhary** [square/okhttp#2327]
Robolectric tests fail with v3.1.1

```
java.lang.IllegalStateException: Unable to extract the trust manager on okhttp3.internal.Platform$Android@56bc3fac, sslSocketFactory is class sun.security.ssl.SSLSocketFactoryImpl
	at okhttp3.OkHttpClient.<init>(OkHttpClient.java:186)
	at okhttp3.OkHttpClient.<init>(OkHttpClient.java:151)
```

Related: #2323

**Comment by Jeff11** [comment]
I got this on an actual Android 5.1.1 device with OkHttp v3.1.1

> java.lang.IllegalStateException: Unable to extract the trust manager on okhttp3.internal.Platform$Android@181dca9b, sslSocketFactory is class my.packagename.MySSLSocketFactory
    at okhttp3.OkHttpClient.<init>(OkHttpClient.java:186)
    at okhttp3.OkHttpClient.<init>(OkHttpClient.java:60)

**Comment by swankjesse** [comment]
@Jeff11 frustratingly, the Java SocketFactory don’t give us a sane mechanism to do proper certificate pinning, and so we hack in tons of nasty reflection. I’m not sure how to proceed with MySSLSocketFactory . . . I’m curious - why do you have a custom SSL socket factory?

**Comment by Jeff11** [comment]
We have our own certificate. The custom ssl socket factory uses a KeyStore with that certificate and falls back to the default KeyStore for other connections.

**Comment by swankjesse** [comment]
@Jeff11 just curious - what happens if you rename the field that contains the fallback SSL socket factory to `delegate` ? We have a special, disgusting hack that’ll make that work possibly, even if that field is private.

**Comment by Jeff11** [comment]
@swankjesse It's working. :D

**Comment by shubhamchaudhary** [comment]
Just to clarify - fixed in v3.1.2

**Comment by IgorGanapolsky** [comment]
I am getting this same error now running a test on emulator.  Using okhttp 3.1.1 and Robolectric 3.0.  What are my options to solve this?
